packages/liveblocks-core/src/lib/position.ts

```diff
@@ -1,133 +1,340 @@
-export const min = 32;
-export const max = 126;
+/**
+ * Positions, aka the Pos type, are efficient encodings of "positions" in
+ * a list, using the following printable subset of the ASCII alphabet:
+ *
+ *    !"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz{|}~
+ *   ^                                                                                             ^
+ *   Lowest digit                                                                      Highest digit
+ *
+ * Each Pos is a sequence of characters from the above alphabet, conceptually
+ * codifying a floating point number 0 < n < 1. For example, the string "31007"
+ * would be used to represent the number 0.31007, except that this
+ * representation uses base 96.
+ *
+ *   0 ≃ ' '  (lowest digit)
+ *   1 ≃ '!'
+ *   2 ≃ '"'
+ *   ...
+ *   9 ≃ '~'  (highest digit)
+ *
+ * So think:
+ *   '!'    ≃ 0.1
+ *   '"'    ≃ 0.2
+ *   '!"~'  ≃ 0.129
+ *
+ * Three rules:
+ * - All "characters" in the string should be valid digits (from the above
+ *   alphabet)
+ * - The value 0.0 is not a valid Pos value
+ * - A Pos cannot have trailing "zeroes"
+ *
+ * This representation has the following benefits:
+ *
+ * 1. It's always possible to get a number that lies before, after, or between
+ *    two arbitrary Pos values.
+ * 2. Pos values can be compared using normal string comparison.
+ *
+ * Some examples:
+ * - '!'  < '"'   (like how .1  < .2)
+ * - '!'  < '~'   (like how .1  < .9)
+ * - '!!' < '!~'  (like how .11 < .19)
+ * - '~!' < '~~'  (like how .91 < .99)
+ * - '~'  < '~!'  (like how .9  < .91)
+ * - '!!' < '!O'  (like how .1  < .5)
+ * - '!O' < '!~'  (like how .5  < .9)
+ *
+ */
+import type { Brand } from "./utils";
 
-export function makePosition(before?: string, after?: string): string {
-  // Between
-  if (before !== undefined && after !== undefined) {
-    return pos(makePositionFromCodes(posCodes(before), posCodes(after)));
-  }
+/**
+ * A valid/verified "position" string. These values are used as "parentKey"s by
+ * LiveList children, and define their relative ordering.
+ */
+export type Pos = Brand<string, "Pos">;
 
-  // Insert at the end
-  else if (before !== undefined) {
-    return getNextPosition(before);
-  }
+const minCode = 32;
```

Comment:
sorry to nit pick, but this const won't change, so maybe rename to `MIN_CODE`, also maybe add a comment that this references an ASCII code. `MAX_CODE` as well.

Reply:
Agreed! Fixed in 3b9d843f1247aaf39e41384cea14bab22b541ddf.

Reply:
I like my constants when they SCREAM at me